unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/BuildModeHUD/Scripts/BuildModeHUDView.cs

```diff
@@ -0,0 +1,246 @@
+
+        this.publishPopupController = publishPopupController;
+        this.publishPopupController.Initialize(publishPopupView);
+
+        this.dragAndDropSceneObjectController = dragAndDropSceneObjectController;
+        this.dragAndDropSceneObjectController.Initialize(dragAndDropSceneObjectView);
+
+        this.publishBtnController = publishBtnController;
+        this.publishBtnController.Initialize(publishBtnView, tooltipController);
+
+        this.inspectorBtnController = inspectorBtnController;
+        this.inspectorBtnController.Initialize(inspectorBtnView, tooltipController);
+
+        this.catalogBtnController = catalogBtnController;
+        this.catalogBtnController.Initialize(catalogBtnView, tooltipController);
+
+        this.inspectorController = inspectorController;
+        this.inspectorController.Initialize(inspectorView);
+
+        this.topActionsButtonsController = topActionsButtonsController;
+        this.topActionsButtonsController.Initialize(topActionsButtonsView, tooltipController);
+    }
+
+    private void OnDestroy()
+    {
+        tooltipController.Dispose();
+        quickBarController.Dispose();
+        sceneCatalogController.Dispose();
+        entityInformationController.Dispose();
+        firstPersonModeController.Dispose();
+        shortcutsController.Dispose();
+        publishPopupController.Dispose();
+        dragAndDropSceneObjectController.Dispose();
+        publishBtnController.Dispose();
+        inspectorBtnController.Dispose();
+        catalogBtnController.Dispose();
+        inspectorController.Dispose();
+        topActionsButtonsController.Dispose();
+    }
+
+    public void PublishStart()
+    {
+        publishPopupController.PublishStart();
+    }
+
+    public void PublishEnd(string message)
+    {
+        publishPopupController.PublishEnd(message);
+    }
+
+    public void SetPublishBtnAvailability(bool isAvailable)
+    {
+        publishBtnController.SetInteractable(isAvailable);
+    }
+
+    public void RefreshCatalogAssetPack()
+    {
+        sceneCatalogController.RefreshAssetPack();
+    }
+
+    public void RefreshCatalogContent()
+    {
+        sceneCatalogController.RefreshCatalog();
+    }
+
+    public void SetVisibilityOfCatalog(bool isVisible)
+    {
+        if (isVisible)
+            sceneCatalogController.OpenCatalog();
+        else
+            sceneCatalogController.CloseCatalog();
+    }
+
+    public void SetVisibilityOfSceneInfo(bool isVisible)
+    {
+        if (isVisible)
+            inspectorController.sceneLimitsController.Enable();
+        else
+            inspectorController.sceneLimitsController.Disable();
+    }
+
+    public void SetVisibilityOfControls(bool isVisible)
+    {
+        shortcutsController.SetActive(isVisible);
+    }
+
+    public void SetVisibilityOfExtraBtns(bool isVisible)
+    {
+        topActionsButtonsController.extraActionsController.SetActive(isVisible);
```

Comment:
Can we have a call in the TopActionButtonController so we don't need to access a controller inside a controller?

Reply:
Done in https://github.com/decentraland/explorer/pull/2103/commits/492d127934ec31b489b9b9d5c6f63ea6ac58a397 Thanks!